.github/workflows/codeql.yml

```diff
@@ -0,0 +1,57 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "7.0", "3.13", "3.14", "3.15", "3.16" ]
+  pull_request:
+    branches: [ "7.0" ]
+  schedule:
+    - cron: "2 10 * * 0"
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
```

Comment:
Does anyone know what "security-events" does, and do we want to give it permissions?

Reply:
`security-events` is required for all CodeQL builds, the others only apply to private repo's.